### DISCUSSION TITLE
TopoFGB? FlatTopoBuf?

### DISCUSSION THREAD

**Discussion by apjoseph** [flatgeobuf/flatgeobuf#390] [Q&A]
I am very impressed by FlatGeoBuf. [This example is mindblowing](https://flatgeobuf.org/examples/maplibre/large.html). For dense fine scale datasets (like parcels, census blocks etc) TopoJSON is also highly compelling as it greatly reduces the size of datasets with shared boundaries.

In my mind, FlatGeoBuf with Topology support seems like the ideal cloud format for large datasets with many shared boundaries. I plan to experiment adding TopoJSON attributes to JSON fields to a FlatGeoBuf dataset  

Are there any limitations/optimizations in FlatGeoBuf that would make this approach redundant or inefficient?

I am also curious if any thought has been given to a FlatTopoBuf format.

**Comment by bjornharrtell** [comment]
I'm quite into topology stuff but haven't considered it in relation to FlatGeobuf except as a transport/intermediate representation of primitive elements like nodes and edges.

I don't think the TopoJSON concept as "extension" to GeoJSON will work for FlatGeobuf. TopoJSON introduces new top level entity "topology" and mandatory "arcs" property so the "extension" is more like "using elements of" for a separate specification. I think a specialized topological binary format needs to be engineered from scratch, but I'm not sure demand/interest is enough for anything to be become generally useful at this point.
